scipy/optimize/_optimize.py

```diff
@@ -1056,14 +1057,31 @@ def check_grad(func, grad, x0, *args, epsilon=_epsilon,
         using `func`. By default it is ``'all'``, in which case, all
         the one hot direction vectors are considered to check `grad`.
         If `func` is a vector valued function then only ``'all'`` can be used.
-    seed : {None, int, `numpy.random.Generator`, `numpy.random.RandomState`}, optional
-        If `seed` is None (or `np.random`), the `numpy.random.RandomState`
-        singleton is used.
-        If `seed` is an int, a new ``RandomState`` instance is used,
-        seeded with `seed`.
-        If `seed` is already a ``Generator`` or ``RandomState`` instance then
-        that instance is used.
-        Specify `seed` for reproducing the return value from this function.
+    rng : {None, int, `numpy.random.Generator`, `numpy.random.RandomState`}, optional
```

Comment:
I didn't use `replace_doc` in the decorator because there's extra information at the bottom of the docstring.

Reply:
Maybe I'll add a rule to the decorator to replace only the paragraph before a blank line?

Reply:
I think that makes sense